service/src/integrationTest/kotlin/fi/espoo/evaka/invoicing/service/InvoiceGeneratorIntegrationTest.kt

```diff
@@ -3545,6 +3548,73 @@ class InvoiceGeneratorIntegrationTest : PureJdbiTest() {
         }
     }
 
+    @Test
+    fun `invoice corrections are applied to invoices when generation is done multiple times`() {
+        val period = DateRange(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 1, 31))
+        db.transaction(insertChildParentRelation(testAdult_1.id, testChild_1.id, period))
+        val decisions = listOf(
+            createFeeDecisionFixture(
+                FeeDecisionStatus.SENT,
+                FeeDecisionType.NORMAL,
+                period,
+                testAdult_1.id,
+                listOf(
+                    createFeeDecisionChildFixture(
+                        childId = testChild_1.id,
+                        dateOfBirth = testChild_1.dateOfBirth,
+                        placementUnitId = testDaycare.id,
+                        placementType = PlacementType.DAYCARE,
+                        serviceNeed = snDaycareFullDay35.toFeeDecisionServiceNeed(),
+                        baseFee = 28900,
+                        fee = 28900
+                    )
+                )
+            )
+        )
+        insertDecisionsAndPlacements(decisions)
+        db.transaction {
+            it.insertTestInvoiceCorrection(
+                DevInvoiceCorrection(
+                    headOfFamilyId = testAdult_1.id,
+                    childId = testChild_1.id,
+                    amount = 1,
+                    unitPrice = -28900,
+                    period = FiniteDateRange(LocalDate.of(2018, 12, 1), LocalDate.of(2018, 12, 31)),
+                    unitId = testDaycare.id,
+                    product = productProvider.mapToProduct(PlacementType.DAYCARE),
+                    description = "",
+                    note = ""
+                )
+            )
+        }
+
+        db.transaction { generator.createAndStoreAllDraftInvoices(it, period) }
+        db.transaction { generator.createAndStoreAllDraftInvoices(it, period) }
```

Comment:
Why is this doubled?

Reply:
The test name was supposed to explain it, but I guess asserting twice would be a good idea 😄